packages/core/src/components/icon/icon.tsx

```diff
@@ -13,16 +13,23 @@ import { Classes, IIntentProps, IProps } from "../../common";
 export { IconName };
 
 export interface IIconProps extends IIntentProps, IProps {
+    /** This component does not support custom children. Use the `icon` prop. */
+    children?: never;
+
     /**
      * Color of icon. Equivalent to setting CSS `fill` property.
      */
     color?: string;
 
     /**
-     * Name of the icon (with or without `"pt-icon-"` prefix).
-     * If omitted or `undefined`, this component will render nothing.
+     * Name of a Blueprint UI icon, or an icon element, to render.
+     *
+     * - If given a string (`IconName` is a TypeScript union of all defined icon names),
+     * that icon will be rendered as an `<svg>` with `<path>` tags.
+     * - If given a `JSX.Element`, that element will be rendered and _all other props on this component are ignored._
+     * - If omitted or falsy, this component will render nothing.
      */
-    iconName?: LegacyIconName;
+    icon: LegacyIconName | JSX.Element | false | null | undefined;
```

Comment:
this is pretty unintuitive, so you should leave clearer documentation on why would a user want to use this with `JSX.Element` since it's just a pass-through render. the short answer is: __they wouldn't__. it's only there to make our codebase cleaner since `<Icon>` is used everywhere.